File: 排序/例题/2.自定义比较器/python/main.py
from functools import cmp_to_key
from typing import List


class Pair:
    def __init__(self, height: int, name: str):
        self.height = height
        self.name = name


class Solution:

    def sortPeople(self, names: List[str], heights: List[int]) -> List[str]:
        n = len(names)
        ans = []
        list1 = []
        for i in range(n):
            list1.append(Pair(heights[i], names[i]))

        def cmp(a, b):
            return b.height - a.height

        list2 = sorted(list1, key=cmp_to_key(cmp))
        for i in range(n):
            ans.append(list2[i].name)
        return ans


if __name__ == '__main__':
    s = Solution()
    s.sortPeople(["Mary", "John", "Emma"], [180, 165, 170])

File: 排序/例题/2.自定义比较器/java/Main.java
import java.util.*;

// 1:无需package
// 2: 类名必须Main, 不可修改
public class Main {

    public class Pair{
        String name;
        int height;
        public Pair(String name,int height){
            this.name = name;this.height=height;
        }
    }
    public String[] sortPeople(String[] names, int[] heights) {
        int n = names.length;
        String[] ans = new String[n];
        List<Pair> list = new ArrayList<>();
        for(int i=0;i<n;i++){
            list.add(new Pair(names[i],heights[i]));
        }
//        Collections.sort(list,(a,b)-> b.height-a.height); lambda表达式写法
        Collections.sort(list, new Comparator<Pair>() {
            @Override
            public int compare(Pair o1, Pair o2) {
                return o2.height-o1.height;
            }
        });//常规写法
        for(int i=0;i<n;i++){
            ans[i] = list.get(i).name;
        }
        return ans;
    }

    public static void main(String[] args) {
        Main main = new Main();
        String[] names = new String[]{"Mary","John","Emma"};
        int[] heights = new int[]{180,165,170};
        main.sortPeople(names,heights);
    }
}